src/main/java/org/checkerframework/specimin/SpeciminRunner.java

```diff
@@ -191,20 +187,20 @@ public static void performMinimization(
           continue;
         }
       }
-
       Path targetOutputPath = Path.of(outputDirectory, target.getKey());
-      // Create any parts of the directory structure that don't already exist.
-      Path dirContainingOutputFile = targetOutputPath.getParent();
-      // This null test is very defensive and might not be required? I think getParent can
-      // only return null if its input was a single element path, which targetOutputPath
-      // should not be unless the user made an error.
-      if (dirContainingOutputFile != null) {
-        Files.createDirectories(dirContainingOutputFile);
-      }
       try {
-        Files.write(
-            targetOutputPath,
-            LexicalPreservingPrinter.print(target.getValue()).getBytes(StandardCharsets.UTF_8));
+        // Create any parts of the directory structure that don't already exist.
+        Path dirContainingOutputFile = targetOutputPath.getParent();
+        // This null test is very defensive and might not be required? I think getParent can
+        // only return null if its input was a single element path, which targetOutputPath
+        // should not be unless the user made an error.
+        if (dirContainingOutputFile != null) {
+          Files.createDirectories(dirContainingOutputFile);
```

Comment:
Why were these lines moved inside the `try` statement? It's good style to keep the size of `try` blocks to a minimum.